charts/IndicatorDropdown.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from "react"
+import AsyncSelect from "react-select/async"
+import { ValueType, SingleValueProps } from "react-select"
+import { bind } from "decko"
+
+import {
+    ExplorerViewContext,
+    ExplorerViewContextType
+} from "./ExplorerViewContext"
+import { observable, action, runInAction } from "mobx"
+import { Indicator } from "./Indicator"
+import { observer } from "mobx-react"
+
+export interface IndicatorDropdownProps {
+    placeholder: string
+    selectedId?: number
+    onChangeId: (id: number) => void
+}
+
+@observer
+export class IndicatorDropdown extends React.Component<IndicatorDropdownProps> {
+    static contextType = ExplorerViewContext
+    context!: ExplorerViewContextType
+
+    static defaultProps = {
+        placeholder: "Type to search..."
+    }
+
+    @observable.ref currentIndicator?: Indicator | null
```

Comment:
TypeScript question: since this is marked optional with `?`, isn't the `| null` redundant?

Reply:
Without `| null` you can't later do:
```js
this.currentIndicator = null
```
That will be a type error. You can still set it to `undefined` though... I chose `null` because `undefined` is set while the indicator is still loading.

Reply:
I see, the `?` only allows undefined, not null. Didn't realize that